README.adoc

```diff
@@ -0,0 +1,247 @@
+  alternative, but given you have already been using Pub/Sub, it is highly
+  likely that you already sold your soul to some other Google Cloud services
+  too. Hence, no need to introduce an extra dependency.] for authentication
+
+- https://github.com/FasterXML/jackson-databind[Jackson]footnote:[https://github.com/googleapis/google-api-java-client[Google
+  APIs Client Library] already depends on this library.] for JSON serialization
+
+- http://micrometer.io/[Micrometer] for metrics
+
+== Getting Started
+
+You first need to add the `reactor-pubsub` artifact into your list of
+dependencies:
+
+```xml
+<dependency>
+    <groupId>com.vlkan</groupId>
+    <artifactId>reactor-pubsub</artifactId>
+    <version>${reactor-pubsub.version}</version>
+</dependency>
+```
+
+In order to instantiate a publisher and/or subscriber, you need to have certain
+basic components in place:
+
+- `ScheduledExecutorService` and `Scheduler` for scheduling asynchronous tasks
+- `PubsubAccessTokenCache` for authenticating HTTP requests
+- `MeterRegistry` for metrics
+- `HttpClient` for HTTP requests
+- `PubsubClient` for Pub/Sub HTTP requests
+
+```java
+// Create the executor that Reactor will subscribe on.
+BoundedScheduledThreadPoolExecutor executorService =
+        new BoundedScheduledThreadPoolExecutor(
+                100, new ScheduledThreadPoolExecutor(8));
+Scheduler scheduler = Schedulers.fromExecutorService(executorService);
+
+// Create the Pub/Sub access token cache.
+PubsubAccessTokenCache accessTokenCache = PubsubAccessTokenCache
+        .builder()
+        .setExecutorService(executorService)
+        .setCredentials("awesome-credentials")
+        .build();
+
+// Create a meter registry for metrics.
+MeterRegistry meterRegistry = new SimpleMeterRegistry();
+
+// Create the Reactor Netty HTTP client.
+HttpClient httpClient = HttpClient.create();
+
+// Create the Pub/Sub client.
+PubsubClient client = PubsubClient
+        .builder()
+        .setAccessTokenCache(accessTokenCache)
+        .setHttpClient(httpClient)
+        .setMeterRegistry(meterRegistry)
+        .build();
+```
+
+You can create a publisher and publish a message as follows:
+
+```java
+// Create the publisher.
+String projectName = "awesome-project";
+PubsubPublisherConfig publisherConfig = PubsubPublisherConfig
+        .builder()
+        .setProjectName(projectName)
+        .setTopicName("awesome-topic")
+        .build();
+PubsubPublisher publisher = PubsubPublisher
+        .builder()
+        .setClient(client)
+        .setConfig(publisherConfig)
+        .build();
+
+// Publish a message.
+publisher
+        .publish(new PubsubPublishRequest(
+                Collections.singletonList(
+                        new PubsubDraftedMessage(
+                                "Yolo like no is watching you!"
+                                        .getBytes(StandardCharsets.UTF_8)))))
+        .subscribeOn(scheduler)
```

Comment:
Why `subscribeOn` is required here? It is rather rare to have this API in the reactive libraries. If you know upfront that `publish` is blocking (although... why?), it is better to require a scheduler to be set when you construct `PubsubPublisher`

Reply:
Indeed, `subscribeOn()` is unnecessary here -- I will remove it.

Reply:
I took an extra step and made `scheduler` optional.